### ISSUE TITLE
Order of components in a Dynamic zone changes after upgrading to 4.5.2

### ISSUE THREAD

**Issue by mirkooom** [strapi/strapi#15035]
Hello,

I upgraded Strapi to the latest version 4.5.2 and i noticed that the components inside the dynamic zone of a collection changed the order.
Both the admin panel and API response are affected.
Rolling back to 4.4.3 restores the original order.

### Screenshots
**Strapi 4.4.3**
<img width="770" alt="strapi_4_4_3" src="https://user-images.githubusercontent.com/119406113/204501879-6ef87abd-ebea-451a-892b-8e580d1365f7.png">

**Strapi 4.5.2**
<img width="751" alt="strapi_4_5_2" src="https://user-images.githubusercontent.com/119406113/204501793-00f03edf-166d-416a-b2c3-2ef1b010f21c.png">

### Expected behavior
The order of the components in a dynamic zone should not change after upgrading

### System Information
Node.js version: v14.19.1
NPM version: v9.1.2
Strapi version: 4.5.2
Database: MySQL 5.7.34
Operating system: Reproduced on both macOS and Linux

**Comment by ronronscelestes** [comment]
hi @mirkooom 
I am not able to reproduce your issue so far 😕

**4.4.3**
<img width="1119" alt="Screenshot 2022-12-09 at 15 24 12" src="https://user-images.githubusercontent.com/71838159/206724195-29415085-0957-4574-b091-1be5f118da7f.png">

**4.5.2**
<img width="1180" alt="Screenshot 2022-12-09 at 15 26 41" src="https://user-images.githubusercontent.com/71838159/206724258-ebffd720-3921-4060-bed5-4cb30038fb17.png">

a few things we could try:
- could you try to revert local version back to 4.4.3, add data & save it, then go back to most recent version (4.5.3) and see effect?
- while doing it could you take a screenshot of the responses (network tab) of both versions?
- could you also share a bit more about your schema?

**Comment by mirkooom** [comment]
I investigated a bit. The problem comes from the column "order" in the table <collection>_components. 

i've been able to replicate the issue in a fresh install.

1. create a component TestComponent
2. create a collection Pages
3. add to Pages the component TestComponent as repeatable component field
4. add to Pages a Dynamic zone with TestComponent component
![Schermata 2022-12-12 alle 15 37 07](https://user-images.githubusercontent.com/119406113/207072599-b1eb6230-98e5-4590-b5aa-bd161c3d2dfc.png)
6. create an entry in the collection and populate both DynamicZone and repeatable component => in the DB table pages_components the column order is populated correctly
7. create an entry in the collection and populate both DynamicZone and repeatable component (more than one entry) => in the DB table pages_components the column order is populated for the current entry, but got set to zero for the first one
<img width="508" alt="Schermata 2022-12-12 alle 15 38 35" src="https://user-images.githubusercontent.com/119406113/207072936-60023468-7a7d-4b14-9863-15efd11c98cd.png">

So basically the second entry save erases the order of other entities

**Comment by joshuaellis** [comment]
@nathan-pichon would we be able to hand this over to either yourself or @Marc-Roig? Seems like a it a BE issue

Cc @ronronscelestes
